include/envoy/http/header_map.h

```diff
@@ -385,7 +385,13 @@ class HeaderEntry {
   virtual void setReference##name(absl::string_view value) PURE;                                   \
   virtual void set##name(absl::string_view value) PURE;                                            \
   virtual void set##name(uint64_t value) PURE;                                                     \
-  virtual size_t remove##name() PURE;
+  virtual size_t remove##name() PURE;                                                              \
+  absl::string_view get##name##Value() const {                                                     \
+    if (name() != nullptr) {                                                                       \
```

Comment:
For further safety, should we deprecate the original name() accessor?  Future PR is fine.

Reply:
I don't think we can completely remove it, since there are some sites that do need to understand presence, but we could replace it with a bool accessor to check for presence potentially?

Reply:
Yeah, worth a thought.  I think wed need to clean up all our current uses (which I had already planned) but also give folks a warning and a few months to update theirs as well.